src/components/Form/Form.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { PropsWithChildren, useEffect } from 'react';
+
+export interface Props extends PropsWithChildren {
+  onSubmit: () => void;
+}
+
+function Form({ children, onSubmit }: Props) {
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Enter') {
+        event.preventDefault();
+        onSubmit();
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
```

Comment:
ok, do we really need this to make it work? should it be default behavior of a form to capture the onSubmit event of its children? pls double check if this is really needed.

Reply:
@robdembitel I checked it again, but by pressing enter by default it does not submit the form. we need to check if user presses enter and then submit using useEffect.